okapi-core/src/main/java/org/folio/okapi/service/impl/DockerModuleHandle.java

```diff
@@ -237,15 +242,51 @@ private void logHandler(Buffer b) {
     return getUrl("/images/" + image + "/json");
   }
 
-  private Future<Void> pullImage() {
-    logger.info("pull image {}", image);
-    return postUrlJson("/images/create?fromImage=" + image, "pullImage", "")
-        .mapEmpty();
+  Future<Void> pullImage() {
+    if (dockerRegistries == null) {
+      logger.info("pull image {}", image);
+      return postUrlJson("/images/create?fromImage=" + image, null, "pullImage", "")
+          .mapEmpty();
+    }
+    logger.info("pull Image using dockerRegistries");
+    Future<Boolean> future = Future.succeededFuture(Boolean.FALSE);
```

Comment:
You can replace the Boolean by future.recover:
```
Future<Void> future = Future.failedFuture("");
for (int i = 0; i < dockerRegistries.size(); i++) {
  ...
  future = future.recover(failure -> {
    ...
    return postUrlJson("/images/create?fromImage=" + prefix + image, authObject, "pullImage", "");
  });
}
return future;
```
This has the additional benefit of returning the failure details of the last registry entry.

Reply:
Much better. Thx